packages/td-tools/src/thing-model-helpers.ts

```diff
@@ -625,7 +625,7 @@ export class ThingModelHelpers {
         keys = keys.map((el) => el.replace("{{", "").replace("}}", ""));
         let isValid = true;
         let errors;
-        if (keys && keys.length > 0 && (map === undefined || map === null)) {
+        if (keys !== undefined && keys.length > 0 && (map === undefined || map === null)) {
```

Comment:
With `foo == null`, you can actually check for `undefined` and `null` at the same time (from my knowledge, this is one of the few cases where `==` is valid to use).

```suggestion
        if ((keys ?? []).length > 0 && (map == null)) {
```